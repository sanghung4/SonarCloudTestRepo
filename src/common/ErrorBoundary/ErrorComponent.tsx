import {
  Box,
  Button,
  Container,
  Grid,
  Hidden,
  Link,
  Typography,
  styled,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink, Redirect, useLocation } from 'react-router-dom';

import leak_bg from 'images/leak_bg.svg';
import leak from 'images/leak.svg';
import stacked_boxes from 'images/stacked_boxes.svg';
import useDocumentTitle from 'hooks/useDocumentTitle';

const Img = styled('img')(({ theme }) => ({
  [theme.breakpoints.up('sm')]: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)'
  }
}));

export enum ErrorTypes {
  NOT_FOUND = 'NOT_FOUND',
  BRANCH_ERROR = 'BRANCH_ERROR',
  OTHER = 'OTHER' // Unused in prod but added for test coverage
}

export type ErrorState = {
  errorType?: ErrorTypes;
};

function ErrorComponent() {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  const { state } = useLocation<ErrorState>();
  useDocumentTitle(t('common.errorTitle'));

  const translationMap = {
    [ErrorTypes.NOT_FOUND]: {
      header: t('product.notFound'),
      body: t('product.notFoundDescription'),
      actionText: t('common.backToHome'),
      actionLink: '/'
    },
    [ErrorTypes.BRANCH_ERROR]: {
      header: t('branch.notFound'),
      body: t('branch.notFoundDescription'),
      actionText: 'Contact Support',
      actionLink: '/support'
    },
    [ErrorTypes.OTHER]: {
      header: t('common.leak'),
      body: t('common.tryGoingBack'),
      actionText: t('common.backToHome'),
      actionLink: '/'
    }
  };
  const isNotFound = state?.errorType === ErrorTypes.NOT_FOUND;
  const isBranchError = state?.errorType === ErrorTypes.BRANCH_ERROR;

  return state?.errorType ? (
    <Box
      bgcolor={isSmallScreen ? 'transparent' : 'common.white'}
      display="flex"
      flex="1"
      mx={isSmallScreen ? 6 : 0}
    >
      <Container maxWidth="md" sx={{ flex: 1, display: 'flex !important' }}>
        <Grid
          container
          spacing={isSmallScreen ? 0 : 6}
          direction={isSmallScreen ? 'column' : 'row'}
        >
          <Hidden mdDown>
            <Grid item xs={12} md={6} position="relative">
              <Img src={leak_bg} alt="leaky pipe background" />
              {isNotFound || isBranchError ? (
                <Img src={stacked_boxes} alt="stacked boxes" />
              ) : (
                <Img src={leak} alt="leaky pipe" />
              )}
            </Grid>
          </Hidden>
          <Grid
            container
            item
            xs={12}
            md={5}
            direction="column"
            justifyContent="center"
            alignItems="center"
          >
            <Hidden mdUp>
              {isNotFound || isBranchError ? (
                <Box
                  component="img"
                  src={stacked_boxes}
                  alt="stacked boxes"
                  mb={5}
                  maxWidth={170}
                  height="auto"
                />
              ) : (
                <Box component="img" src={leak} alt="leaky pipe" mb={5} />
              )}
            </Hidden>
            {!isNotFound && !isBranchError && (
              <Typography
                variant={isSmallScreen ? 'h2' : 'h1'}
                color="primary02.main"
                fontWeight={400}
              >
                {t('common.ohNo')}
              </Typography>
            )}
            <Typography
              variant={isSmallScreen ? 'h5' : 'h3'}
              data-testid="product-not-found"
              pt={isSmallScreen ? 0.625 : 0.75}
            >
              {translationMap[state.errorType].header}
            </Typography>
            <Typography
              variant={isSmallScreen ? 'body1' : 'h5'}
              align="center"
              py={isSmallScreen ? 2 : 4}
              fontWeight={400}
            >
              {translationMap[state.errorType].body}
            </Typography>
            <Grid
              container
              spacing={isSmallScreen ? 2 : 3}
              flexDirection={
                isSmallScreen || isNotFound || isBranchError
                  ? 'column-reverse'
                  : 'row'
              }
              alignItems="center"
              px={4}
              pt={isSmallScreen ? 0.5 : 0}
            >
              <Grid item xs={12} md={isNotFound ? 6 : 12} minWidth={150}>
                <Link
                  to={translationMap[state.errorType].actionLink}
                  sx={{ display: 'flex' }}
                  component={RouterLink}
                  underline="none"
                >
                  <Button
                    variant="primary"
                    fullWidth
                    data-testid="back-to-home-button"
                  >
                    {translationMap[state.errorType].actionText}
                  </Button>
                </Link>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Container>
    </Box>
  ) : (
    <Redirect to="/" />
  );
}

export default ErrorComponent;
