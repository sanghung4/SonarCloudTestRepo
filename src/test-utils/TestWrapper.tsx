import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
