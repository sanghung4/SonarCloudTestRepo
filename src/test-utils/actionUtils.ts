import { fireEvent, findByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export const fillFormTextInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const fillMaskedFormTextInput = async (
  testId: string,
  value: string
) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await userEvent.type(textInput, value);
  });
};

export const clickButton = async (testId: string) => {
  await act(async () => {
    const button = await findByTestId(document.body, testId);
    await fireEvent.click(button);
  });
};

export const clickRadio = async (testId: string, optionIndex: number) => {
  await act(async () => {
    const radio = await findByTestId(
      document.body,
      `${testId}-radio-${optionIndex}`
    );
    await userEvent.click(radio);
  });
};

export const fillFormSelectInput = async (testId: string, value: string) => {
  await act(async () => {
    const selectInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(selectInput, { target: { value } });
    await fireEvent.keyDown(window, { key: 'Enter' });
  });
};

export const focusInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.focus(input);
  });
};

export const changeInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, testId);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const blurInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.blur(input);
  });
};

export const keyboardDown = async (
  key: string,
  element?: Element | Node | Document | Window
) => {
  await act(async () => {
    await fireEvent.keyDown(element ?? window, { key });
  });
};
