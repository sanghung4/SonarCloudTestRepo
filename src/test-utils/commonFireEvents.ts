import { act, fireEvent } from '@testing-library/react';

export function actClickOnElement(element: HTMLElement) {
  act(() => {
    fireEvent.click(element);
  });
}
export function actUpdateInput(input: HTMLElement, value: string) {
  act(() => updateInput(input, value));
}
export function updateInput(input: HTMLElement, value: string) {
  fireEvent.focus(input);
  fireEvent.change(input, { target: { value } });
  fireEvent.blur(input);
}
