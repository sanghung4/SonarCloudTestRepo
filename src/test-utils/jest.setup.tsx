import React, { JSXElementConstructor, ReactElement, ReactNode } from "react";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { render as jestRender } from "@testing-library/react";

export const mockApollo = jest.mock(
  "@apollo/client/testing",
  () => MockedProvider
);

interface Props {
  children?: ReactNode;
  mocks?: ReadonlyArray<MockedResponse>;
}

function TestWrapper(props: Props) {
  return (
    <MockedProvider mocks={props.mocks ?? []}>{props.children}</MockedProvider>
  );
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined,
  });
}
