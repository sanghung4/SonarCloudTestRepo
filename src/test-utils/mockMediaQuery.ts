import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
