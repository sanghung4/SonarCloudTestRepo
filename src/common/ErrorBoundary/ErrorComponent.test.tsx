import ErrorComponent, {
  ErrorTypes
} from 'common/ErrorBoundary/ErrorComponent';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

type MockRedirectToObj = {
  pathname?: string;
  state?: unknown;
};
type MockRedirectProp = {
  to: string | MockRedirectToObj;
};
type MockState = { errorType?: string };

const mockState: MockState = {
  errorType: ErrorTypes.NOT_FOUND
};
function MockRedirect({ to }: MockRedirectProp) {
  return (
    <span data-testid="redirect">
      {typeof to === 'string' ? to : to.pathname}
    </span>
  );
}

jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    state: mockState
  }),
  Redirect: MockRedirect
}));

const setup = () => {
  const utils = render(<ErrorComponent />);

  const productNotFoundText = utils.getByTestId('product-not-found');
  const backToHomeButton = utils.getByTestId('back-to-home-button');
  return {
    utils,
    productNotFoundText,
    backToHomeButton
  };
};

describe('common - ErrorBoundary - ErrorComponent', () => {
  it('should match snapshot on desktop as NOT_FOUND', () => {
    setBreakpoint('desktop');
    mockState.errorType = ErrorTypes.NOT_FOUND;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile as NOT_FOUND', () => {
    setBreakpoint('mobile');
    mockState.errorType = ErrorTypes.NOT_FOUND;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on desktop as BRANCH_ERROR', () => {
    setBreakpoint('desktop');
    mockState.errorType = ErrorTypes.BRANCH_ERROR;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile as BRANCH_ERROR', () => {
    setBreakpoint('mobile');
    mockState.errorType = ErrorTypes.BRANCH_ERROR;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('Should navigate to Error page with Product Not Found error type', () => {
    setBreakpoint('desktop');
    mockState.errorType = ErrorTypes.NOT_FOUND;
    const { productNotFoundText, backToHomeButton } = setup();

    expect(productNotFoundText).toBeTruthy();
    expect(backToHomeButton).toBeTruthy();
  });

  it('should match snapshot on desktop as OTHER', () => {
    setBreakpoint('desktop');
    mockState.errorType = ErrorTypes.OTHER;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile as OTHER', () => {
    setBreakpoint('mobile');
    mockState.errorType = ErrorTypes.OTHER;
    const {
      utils: { container }
    } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should expect Redirect to be "rendered" as `undefined`', () => {
    setBreakpoint('desktop');
    mockState.errorType = undefined;
    const { getByTestId } = render(<ErrorComponent />);
    expect(getByTestId('redirect')).toHaveTextContent('/');
  });
});
