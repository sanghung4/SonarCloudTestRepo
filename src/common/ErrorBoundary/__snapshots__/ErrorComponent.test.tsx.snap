// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on desktop as BRANCH_ERROR 1`] = `
<div>
  <div
    class="MuiBox-root css-152jcjz"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-honmcm-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-mpgh9h-MuiGrid-root"
        >
          <img
            alt="leaky pipe background"
            class="css-rvrkdc"
            src="leak_bg.svg"
          />
          <img
            alt="stacked boxes"
            class="css-rvrkdc"
            src="stacked_boxes.svg"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-1sf1x69-MuiTypography-root"
            data-testid="product-not-found"
          >
            Unable to load home branch
          </h3>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1gahoiz-MuiTypography-root"
          >
            Please contact support or select a job with a different home branch.
          </h5>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1o335il-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-15sgum0-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/support"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Contact Support
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on desktop as NOT_FOUND 1`] = `
<div>
  <div
    class="MuiBox-root css-152jcjz"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-honmcm-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-mpgh9h-MuiGrid-root"
        >
          <img
            alt="leaky pipe background"
            class="css-rvrkdc"
            src="leak_bg.svg"
          />
          <img
            alt="stacked boxes"
            class="css-rvrkdc"
            src="stacked_boxes.svg"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-1sf1x69-MuiTypography-root"
            data-testid="product-not-found"
          >
            Product not found
          </h3>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1gahoiz-MuiTypography-root"
          >
            Please try searching again or return to the home page.
          </h5>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1o335il-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-djd5wm-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Back to Home
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on desktop as OTHER 1`] = `
<div>
  <div
    class="MuiBox-root css-152jcjz"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-honmcm-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-mpgh9h-MuiGrid-root"
        >
          <img
            alt="leaky pipe background"
            class="css-rvrkdc"
            src="leak_bg.svg"
          />
          <img
            alt="leaky pipe"
            class="css-rvrkdc"
            src="leak.svg"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-1iv52y0-MuiTypography-root"
          >
            Oh no!
          </h1>
          <h3
            class="MuiTypography-root MuiTypography-h3 css-1sf1x69-MuiTypography-root"
            data-testid="product-not-found"
          >
            Looks like we have a leak
          </h3>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1gahoiz-MuiTypography-root"
          >
            Try going back to the previous page while we take care of this.
          </h5>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-d8n3rb-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-15sgum0-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Back to Home
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on mobile as BRANCH_ERROR 1`] = `
<div>
  <div
    class="MuiBox-root css-101xdvu"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <img
            alt="stacked boxes"
            class="MuiBox-root css-z296mr"
            src="stacked_boxes.svg"
          />
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1tm0viy-MuiTypography-root"
            data-testid="product-not-found"
          >
            Unable to load home branch
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mbm39a-MuiTypography-root"
          >
            Please contact support or select a job with a different home branch.
          </p>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-roukp-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-15sgum0-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/support"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Contact Support
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on mobile as NOT_FOUND 1`] = `
<div>
  <div
    class="MuiBox-root css-101xdvu"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <img
            alt="stacked boxes"
            class="MuiBox-root css-z296mr"
            src="stacked_boxes.svg"
          />
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1tm0viy-MuiTypography-root"
            data-testid="product-not-found"
          >
            Product not found
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mbm39a-MuiTypography-root"
          >
            Please try searching again or return to the home page.
          </p>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-roukp-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-djd5wm-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Back to Home
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - ErrorBoundary - ErrorComponent should match snapshot on mobile as OTHER 1`] = `
<div>
  <div
    class="MuiBox-root css-101xdvu"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-a9wr92-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-ao3jzh-MuiGrid-root"
        >
          <img
            alt="leaky pipe"
            class="MuiBox-root css-10ib5jr"
            src="leak.svg"
          />
          <h2
            class="MuiTypography-root MuiTypography-h2 css-5ht2wf-MuiTypography-root"
          >
            Oh no!
          </h2>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1tm0viy-MuiTypography-root"
            data-testid="product-not-found"
          >
            Looks like we have a leak
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mbm39a-MuiTypography-root"
          >
            Try going back to the previous page while we take care of this.
          </p>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-roukp-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-15sgum0-MuiGrid-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-q5uxwt-MuiTypography-root-MuiLink-root"
                to="/"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1iivgkm-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-to-home-button"
                  tabindex="0"
                  type="button"
                >
                  Back to Home
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
